test/test_hyperparameters.py

```diff
@@ -328,6 +326,179 @@ def test_normalfloat_to_integer(self):
+        self.assertEqual(f1.log, False)
+        self.assertAlmostEqual(f1.default_value, 2.0)
+        # beta parameters are scaled to the unit hypercube
+        self.assertAlmostEqual(f1.normalized_default_value, 1.0)
+
+        # Test copy
+        copy_f1 = copy.copy(f1)
+
+        self.assertEqual(copy_f1.name, f1.name)
+        self.assertEqual(copy_f1.alpha, f1.alpha)
+        self.assertEqual(copy_f1.beta, f1.beta)
+        self.assertEqual(copy_f1.default_value, f1.default_value)
+
+        f2 = BetaFloatHyperparameter("param", lower=-2.0, upper=2.0, alpha=3.0, beta=1.0, q=0.1)
+        f2_ = BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=3, beta=1, q=0.1)
+        self.assertEqual(f2, f2_)
+
+        self.assertEqual(
+            "param, Type: BetaFloat, Alpha: 3.0 Beta: 1.0, Range: [-2.0, 2.0], Default: 2.0, "
+            "Q: 0.1", str(f2))
+
+        f3 = BetaFloatHyperparameter("param", lower=10**(-5), upper=10.0,
+                                     alpha=6.0, beta=2.0, log=True)
+        f3_ = BetaFloatHyperparameter("param", lower=10**(-5),
+                                      upper=10.0, alpha=6.0, beta=2.0, log=True)
+        self.assertEqual(f3, f3_)
+        self.assertEqual(
+            "param, Type: BetaFloat, Alpha: 6.0 Beta: 2.0, Range: [1e-05, 10.0], Default: 1.0, "
+            "on log-scale", str(f3))
+
+        with self.assertRaises(ValueError):
+            BetaFloatHyperparameter("param", lower=-1, upper=10.0, alpha=6.0, beta=2.0, log=True)
+
+        f4 = BetaFloatHyperparameter("param", lower=1, upper=1000.0,
+                                     alpha=2.0, beta=2.0, log=True, q=1.0)
+        f4_ = BetaFloatHyperparameter("param", lower=1, upper=1000.0,
+                                      alpha=2.0, beta=2.0, log=True, q=1.0)
+
+        self.assertEqual(f4, f4_)
+        self.assertEqual(
+            "param, Type: BetaFloat, Alpha: 2.0 Beta: 2.0, Range: [1.0, 1000.0], Default: 22.0, "
+            "on log-scale, Q: 1.0", str(f4))
+
+        with self.assertRaisesRegex(ValueError, "Illegal default value 0"):
+            BetaFloatHyperparameter("param", lower=1, upper=10.0, alpha=3.0,
+                                    beta=2.0, default_value=0, log=False)
+        with self.assertRaisesRegex(ValueError, "Illegal default value 0"):
+            BetaFloatHyperparameter("param", lower=1, upper=1000.0, alpha=3.0,
+                                    beta=2.0, default_value=0, log=True)
+
+        f5_legal_nolog = BetaFloatHyperparameter(
+            "param", lower=1, upper=10.0, alpha=3.0, beta=2.0, default_value=1, log=True)
+        f5_legal_log = BetaFloatHyperparameter(
+            "param", lower=1, upper=10.0, alpha=3.0, beta=2.0, default_value=1, log=False)
+
+        # Testing that no error is thrown
+        BetaFloatHyperparameter("param", lower=1, upper=1000.0, alpha=2.0, beta=3.0, q=3.0)
+
+        self.assertAlmostEqual(f5_legal_nolog.default_value, 1)
```

Comment:
Maybe we could have a separate test for the default values that checks for log, quantization, and alpha being 1 and beta being one.

Reply:
The same holds for the integer version. Looking at the code, this is really the only thing that is different than the parent class and that needs to be tested thoroughly

Reply:
I think it's up to us to decide whether log should move the default value or not. Personally, I think it shouldn't.

Yeah sure, I can split them up and do some more explicit tests for default values.

Reply:
> I think it's up to us to decide whether log should move the default value or not. Personally, I think it shouldn't.

I think diagree on this one. Let's take the example you did below with a hyperparameter from 1 to 100. In the non-log case the default (in my opinion) should be 50, while in the log case it should be 10. In both cases it would be the middle of the distribution.

Reply:
Oh, on that one I completely agree! I was strictly talking about the case where **the user specifies the default**, which is the case in the example you brought up.